bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java

```diff
@@ -1034,6 +1127,11 @@ public String toString() {
             return;
         }
 
+        if (!waitForWritable(distributionSchedule.getWriteSet(op.getEntryId()), op.getEntryId(),
+                0, waitForWriteSetMs)) {
+            op.allowFastFailOnBlockedServer();
```

Comment:
I would have loved to contain this logic at the channel/PCBC level. But I can live with this. :)

Reply:
PCBC runs in context of different thread (worker pool)